socs/agents/hwp_pmx/agent.py

```diff
@@ -260,6 +298,86 @@ def _stop_acq(self, session, params=None):
         else:
             return False, 'acq is not currently running'
 
+    def initiate_shutdown(self, session, params):
+        """ initiate_shutdown()
+
+        **Task** - Initiate the shutdown of the agent.
+        """
+        self.log.warn("INITIATING SHUTDOWN")
+
+        with self.lock.acquire_timeout(10, job='shutdown') as acquired:
+            if not acquired:
+                self.log.error("Could not acquire lock for shutdown.")
+                return False, "Could not acquire lock."
+
+            self.shutdown_mode = True
+            self.dev.ign_external_voltage()
```

Comment:
Technically this line is unnecessary, as you can shut off the power in both run modes (with or without PID)